wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -727,6 +771,12 @@ private Connection connectInternal(String driverProtocol, HostSpec hostSpec, Pro
         this.staleDnsHelper.getVerifiedConnection(isInitialConnection, this.hostListProviderService,
             driverProtocol, hostSpec, props, connectFunc);
 
+    if (this.keepSessionStateOnFailover) {
+      this.savedReadOnlyStatus = this.savedReadOnlyStatus == null ? conn.isReadOnly() : this.savedReadOnlyStatus;
```

Comment:
Do we update these stored values in case you call setReadOnly() or setAutoCommit() after connection is open?